tracer/src/Datadog.Trace/Activity/OtlpHelpers.cs

```diff
@@ -344,6 +357,61 @@ internal static void AgentStatus2Error<TInner>(TInner activity, Span span)
             }
         }
 
+        /// <summary>
+        /// Iterates through <c>Activity.Events</c> looking for a <see cref="OpenTelemetryException"/> to copy over
+        /// to the <see cref="Span.Tags"/> of <paramref name="span"/>.
+        /// </summary>
+        /// <param name="activity">The <see cref="IActivity"/> to check for the exception event data.</param>
+        /// <param name="span">The <see cref="Span"/> to copy the exception event data to.</param>
+        /// <typeparam name="TInner">
+        /// The type of <c>Activity</c> - note that only <see cref="IActivity5"/> and up have support for events.
+        /// </typeparam>
+        /// <remarks>OpenTelemetry creates these attributes via it's <c>Activity.RecordException</c> function.</remarks>
+        private static void ExtractExceptionAttributes<TInner>(TInner activity, Span span)
+            where TInner : IActivity
+        {
+            // OpenTelemetry stores the exception attributes in Activity.Events
+            // Activity.Events was only added in .NET 5+, which maps to out IActivity5 & IActivity6
+            if (activity is not IActivity5 activity5)
+            {
+                return;
+            }
+
+            foreach (var activityEvent in activity5.Events)
```

Comment:
From what I can tell, the allocation happens when the enumerator returns an object and has to box the struct

Reply:
I think this is not supported right now, but it’s technical possible to create an struct with the same memory layout as the target and just use that. But currently I think it doesn’t pass the ducktype checks.

Reply:
If this is going to be in a hot path. I prefer to do that and avoid the allocations

Reply:
I think since this only currently would check when the span is an error it isn't a hot path yet? 
But if we swap it to check for all spans for exception info then probably would be in the hot path.